src/batched/dense/KokkosBatched_Vector.hpp

```diff
@@ -255,6 +255,18 @@ class ArithTraits<KokkosBatched::Vector<KokkosBatched::SIMD<T>, l>> {
     return val;
   }
 
+  static KOKKOS_FORCEINLINE_FUNCTION val_type conj(const val_type &val) {
+    return val;
+  }
+
+  static KOKKOS_FORCEINLINE_FUNCTION mag_scalar_type abs(const val_type &val) {
+    mag_scalar_type v = ArithTraits<mag_scalar_type>::zero();
+    for (int i = 0; i < l; ++i) {
+      v = KOKKOSKERNELS_MACRO_MAX(v, KOKKOSKERNELS_MACRO_ABS(val[i]));
```

Comment:
That should always be `abs(val[i])` or abs is not working correctly and sometimes return a value that is less than 0?

Reply:
This is `abs()` inside our implementation of `Kokkos::ArithTraits` for `Vector<SIMD<T>>`, which computes: $|V|=max_i |V_i|$  (`v` takes `max` of `abs` of vector values _val[i]_)



Looking at it now, I think maybe I should have checked with you beforehand if that's the correct formula...

Reply:
As discussed, On https://github.com/kokkos/kokkos-kernels/pull/1433/commits/508f253506644ed8634f4a36781e39889784782b I've fixed `abs()` to return vector values (similar to `conj()`) for SIMD vectors.

I've changed the vector comparison in unit tests to go value-by-value, re-introducing `EXPECT_NEAR_KK_REL` for the convenient vector overload. The generic comparison based on `EXPECT_NEAR_KK(v1, v2, tol * Kokkos::max(abs(v1), abs(v2)))` wouldn't work for vectors anyway, since - with `abs()` returning vectors now - we'd be missing `max(vector1, vector2)` that probably can't be defined anyway as vectors have no linear order).

Reply:
Thanks again for pointing this one out!